# HG changeset patch
# User adf88@interia.pl
# Date 1482656360 -3600
#      Sun Dec 25 09:59:20 2016 +0100
# Branch trunk
# Node ID fd244df3071e74c5f400d9a33256d1806c2a19e1
# Parent  fadf958567a2c8370d3f637066c415c74859fdbb
[mq]: fix-clipping-by-window-clip

diff -r fadf958567a2 -r fd244df3071e src/window.cpp
--- a/src/window.cpp	Sat Dec 24 18:45:38 2016 +0000
+++ b/src/window.cpp	Sun Dec 25 09:59:20 2016 +0100
@@ -942,7 +942,7 @@
 				left < w->left + w->width &&
 				top < w->top + w->height) {
 			/* Window w intersects with the rectangle => needs repaint */
-			DrawOverlappedWindow(w, left, top, right, bottom);
+			DrawOverlappedWindow(w, max(left, w->left), max(top, w->top), min(right, w->left + w->width), min(bottom, w->top + w->height));
 		}
 	}
 }
